spark/src/main/java/org/opensearch/sql/spark/dispatcher/InteractiveQueryHandler.java

```diff
@@ -62,7 +63,13 @@ protected JSONObject getResponseFromExecutor(AsyncQueryJobMetadata asyncQueryJob
   @Override
   public String cancelJob(AsyncQueryJobMetadata asyncQueryJobMetadata) {
     String queryId = asyncQueryJobMetadata.getQueryId().getId();
-    getStatementByQueryId(asyncQueryJobMetadata.getSessionId(), queryId).cancel();
+    Statement statement = getStatementByQueryId(asyncQueryJobMetadata.getSessionId(), queryId);
+    if (NOT_CANCELLABLE_STATE.contains(statement.getStatementState())) {
+      throw new IllegalArgumentException(
+          String.format(
+              "can't cancel statement in %s state", statement.getStatementState().getState()));
+    }
+    statement.cancel();
```

Comment:
it's already here 

https://github.com/opensearch-project/sql/blob/main/spark/src/main/java/org/opensearch/sql/spark/execution/statement/Statement.java#L67-L76

Reply:
thanks.